src/Streamly/Data/Serialize/Instances/ByteString.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE UnboxedTuples #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Streamly.Data.Serialize.Instances.ByteString () where
+
+import Streamly.Internal.Data.Serialize (Serialize(..))
+import Data.ByteString.Internal as Strict
+import Data.ByteString.Lazy as Lazy
+import qualified Streamly.Internal.Data.Unbox as Unbox
+import GHC.Exts
+import Foreign.ForeignPtr (withForeignPtr)
+import Foreign.Ptr (plusPtr)
+import GHC.Base (IO(..))
+import Control.Monad (foldM_)
+
+--------------------------------------------------------------------------------
+-- Strict ByteString
+--------------------------------------------------------------------------------
+instance Serialize Strict.ByteString where
+    {-# INLINE size #-}
+    size i (Strict.PS _ _ l) = i + l + size 0 l
```

Comment:
You can use `Unbox.sizeOf (Proxy :: Proxy Int64)` or simply just use `8` instead of `size 0 l`.
```
i + l + 8
```
Or,
```
i + l + Unbox.sizeOf (Proxy :: Proxy Int64)
```